src/test/java/javax/measure/unit/UnitTest.java

```diff
@@ -1,11 +1,134 @@
+    // all failed before the fix was made.
+    @Test
+    public void testParseUnitsWithDivisions() throws Exception {
+
+        SoftAssertions softly = new SoftAssertions();
+
+        for (Map.Entry<String, String> divisionUnit : DIVISION_UNITS.entrySet()) {
+            String parsedUnit = Unit.valueOf(divisionUnit.getKey()).toString();
+            softly.assertThat(parsedUnit).isEqualTo(divisionUnit.getValue());
+
+            // Test it again but replace * with ·.  Example: m/kg*h -> m/kg·h
+            parsedUnit = Unit.valueOf(divisionUnit.getKey().replace("*", "·")).toString();
+            softly.assertThat(parsedUnit).isEqualTo(divisionUnit.getValue());
+        }
+
+        softly.assertAll();
+    }
+
+    private static final Map<String, String> DIVISION_UNITS;
+
+    static {
+        Map<String, String> map = new LinkedHashMap<String, String>();
+
+        // 1 Divide
+        map.put("N/kg", "N/kg");
+
+        // 2 Divides
+        map.put("N/kg/h", "N/(kg·h)");
+        map.put("(N/kg)/h", "N/(kg·h)");
+        map.put("N/(kg/h)", "N·h/kg");
+        map.put("N/(kg)/h", "N/(kg·h)");
+
+        // 1 Divide+ 1 Multiply
+        map.put("N*kg/h", "N·kg/h");
+        map.put("(N*kg)/h", "N·kg/h");
+        map.put("N*(kg/h)", "N·kg/h");
+        map.put("N*(kg)/h", "N·kg/h");
+
+        map.put("N/kg*h", "N·h/kg");
+        map.put("(N/kg)*h", "N·h/kg");
+        map.put("N/(kg*h)", "N/(kg·h)");
+        map.put("N/(kg)*h", "N·h/kg");
+
+        // 3 Divides
+        map.put("N/kg/m/h", "N/(kg·m·h)");
+        map.put("((N/kg)/m)/h", "N/(kg·m·h)");
+        map.put("(N/kg)/(m/h)", "N·h/(kg·m)");
+        map.put("N/(kg/(m/h))", "N·m/(kg·h)");
+
+        map.put("(N/kg)/m/h", "N/(kg·m·h)");
+        map.put("N/(kg/m)/h", "N·m/(kg·h)");
+        map.put("N/kg/(m/h)", "N·h/(kg·m)");
+
+        // 2 Divides + 1 Multiply
+        map.put("N*kg/m/h", "N·kg/(m·h)");
+        map.put("N/kg*m/h", "N·m/(kg·h)");
+        map.put("N/kg/m*h", "N·h/(kg·m)");
+
+        map.put("(N*kg)/m/h", "N·kg/(m·h)");
+        map.put("(N/kg)*m/h", "N·m/(kg·h)");
+        map.put("(N/kg)/m*h", "N·h/(kg·m)");
+
+        map.put("N*(kg/m)/h", "N·kg/(m·h)");
+        map.put("N/(kg*m)/h", "N/(kg·m·h)");
+        map.put("N/(kg/m)*h", "N·m·h/kg");
+
+        map.put("N*kg/(m/h)", "N·kg·h/m");
+        map.put("N/kg*(m/h)", "N·m/(kg·h)");
+        map.put("N/kg/(m*h)", "N/(kg·m·h)");
+
+        // 1 Divide + 2 Multiplies
+        map.put("N/kg*m*h", "N·m·h/kg");
+        map.put("N*kg/m*h", "N·kg·h/m");
+        map.put("N*kg*m/h", "N·kg·m/h");
+
+        map.put("(N/kg)*m*h", "N·m·h/kg");
+        map.put("(N*kg)/m*h", "N·kg·h/m");
+        map.put("(N*kg)*m/h", "N·kg·m/h");
+
+        map.put("N/(kg*m)*h", "N·h/(kg·m)");
+        map.put("N*(kg/m)*h", "N·kg·h/m");
+        map.put("N*(kg*m)/h", "N·kg·m/h");
+
+        map.put("N/kg*(m*h)", "N·m·h/kg");
+        map.put("N*kg/(m*h)", "N·kg/(m·h)");
+        map.put("N*kg*(m/h)", "N·kg·m/h");
+
```

Comment:
The rest of the tests here didn't pass with my previous workaround.  (It wasn't a big concern because units of this structure are unlikely.)  They pass with this fix.